projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3021,8 +3023,12 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
                 return;
             }
 
-            if (this.overlayService.getOverlayById(event.id)?.settings?.outlet === this.outlet &&
-                this.overlayIDs.indexOf(event.id) < 0) {
+            if (overlaySettings?.outlet === this.outlet && this.overlayIDs.indexOf(event.id) === -1) {
+                // do not hide the overlay if it's attached to a row for editing purposes
+                if (overlaySettings.positionStrategy instanceof RowEditPositionStrategy) {
```

Comment:
the RowEditPositionStrategy can be used from other component, because it is public.
The IgxToggleDirective should be extended to return the overlayId and we can check if the rowEditingOverlay overlay id is different than the event.id